### PR TITLE
update tidb version to latest

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
update tidb version after https://github.com/pingcap/tidb/pull/16596 merged

### What is changed and how it works?
update tidb version, so we can handle auto-random config in incremental backup

### Check List 

Tests 

 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - No release note

### CODE REVIEW

**Comment by kennytm** [comment]
Please resolve go.mod conflict.

File: pkg/restore/db.go [hunk 1]
```diff
@@ -123,13 +123,31 @@ func (db *DB) CreateTable(ctx context.Context, table *utils.Table) error {
...
+					zap.String("query", restoreMetaSQL),
+					zap.Stringer("db", table.Db.Name),
+					zap.Stringer("table", table.Info.Name),
+					zap.Error(err))
+				return errors.Trace(err)
+			}
+			restoreMetaSQL = fmt.Sprintf(setValFormat, table.Info.AutoIncID)
 		} else {
 			restoreMetaSQL = fmt.Sprintf(setValFormat, table.Info.AutoIncID)
 		}
```
Comment (kennytm) [inline pkg/restore/db.go:153]: The `else` branch can be removed and the assignment be pulled out of the `if` statement.

File: pkg/restore/db.go [hunk 2]
```diff
@@ -123,13 +123,31 @@ func (db *DB) CreateTable(ctx context.Context, table *utils.Table) error {
...
+				return errors.Trace(err)
+			}
+
 			// trigger cycle round > 0
-			restoreMetaSQL += nextSeqSQL
-			restoreMetaSQL += fmt.Sprintf(setValFormat, table.Info.AutoIncID)
+			err = db.se.Execute(ctx, nextSeqSQL)
+			if err != nil {
+				log.Error("restore meta sql failed",
+					zap.String("query", restoreMetaSQL),
```
Comment (kennytm) [inline pkg/restore/db.go:144]:
```suggestion
					zap.String("query", nextSeqSQL),
```

File: pkg/restore/db.go [hunk 3]
```diff
@@ -123,13 +123,31 @@ func (db *DB) CreateTable(ctx context.Context, table *utils.Table) error {
 				utils.EncloseName(table.Db.Name.O),
 				utils.EncloseName(table.Info.Name.O))
 			if increment < 0 {
-				restoreMetaSQL += fmt.Sprintf(setValFormat, table.Info.Sequence.MinValue)
+				restoreMetaSQL = fmt.Sprintf(setValFormat, table.Info.Sequence.MinValue)
 			} else {
-				restoreMetaSQL += fmt.Sprintf(setValFormat, table.Info.Sequence.MaxValue)
+				restoreMetaSQL = fmt.Sprintf(setValFormat, table.Info.Sequence.MaxValue)
 			}
+			err = db.se.Execute(ctx, restoreMetaSQL)
```
Comment (kennytm) [inline pkg/restore/db.go:130]: let's use a different variable name

**Comment by 3pointer** [comment]
we need cherry-pick this pr manually after TiDB cherry-pick https://github.com/pingcap/tidb/pull/16596 to release-3.1 and release-4.0.

**Pull request merged**